common/lib/container-definitions/package.json

```diff
@@ -74,6 +74,9 @@
     "broken": {
       "InterfaceDeclaration_IRuntime": {
         "forwardCompat": false
+      },
```

Comment:
Curious, why do you need to change it? Seems like interface is not changed

Reply:
The `storage` property on `IContainerContext` is `IDocumentStorageService`, so the back compat test fails since the `IDocumentStorageService` signatures aren't the same

Reply:
Hm, this feels weird as I'd not expect version check to be nested like that. If it goes recursive, a change anywhere will cause a lot of types to be different. @anthony-murphy, any chance you can glance?

Reply:
@vladsud Every time I build it finds new things broken with back compat. Because it seems to be recursive, it will be a lot bigger of a change than I anticipated. I'm currently very tempted to revert my commit that removed this method.

Reply:
This feels wrong. Can you please ping Tony? I do not think it should be recursive.
Maybe other way to say it - we need to be able to make changes like that. Maybe here we can punt, but it would be nuts if we can't change interfaces rather easily.

Reply:
this is the way it currently works, and represents the real break you've made. The type tests don't do anything fancy, they just check if the old type can be used in place of the new type and vis versa via the typescript compiler.

Reply:
to me this points to issues is our code. why do we expose the same thing in so many places?

Reply:
@anthony-murphy Take `IProvideContainerRuntime` for example:
1. `IProvideContainerRuntime` has a property with type `IContainerRuntime`
2. `IContainerRuntime` has a property with type `IDocumentStorageService`
3. And thus, back compat for `IProvideContainerRuntime` breaks

Changing low-level interfaces seems to spread quite a bit even though no one is really consuming this `write` method. There are a number of places that implement the `IDocumentStorageService`, so the breaking spreads very quickly (which could be expected).

Reply:
I'm sure it's not the last time we will hit issues like that. It feels like one of the two should be possible:
1. A way for developers to ignore all errors (without updating 100 places)
2. A script that developer can run to auto-generate package.json changes, similar to api.md generation process.

I personally would favor # 2, and I find I need it even for trivial compat changes. It feels like it's in same bucket as linting/formatting changes - it's so much easier to work in Bohemia repo where changes are done automatically, and dev responsibility is to review them.